bears/python/PEP8NotebookBear.py

```diff
@@ -0,0 +1,130 @@
+import autopep8
+from copy import copy
+import nbformat
+
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from coalib.bears.LocalBear import LocalBear
+from coalib.bears.requirements.PipRequirement import PipRequirement
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.settings.Setting import typed_list
+
+
+class NotebookHandler:
+    """Utility class for Jupyter Notebook handling."""
+
+    def __init__(self, filename, file):
+        """
+        Constructs a notebook handler to deal with NotebookNode objects.
+        A NotebookNode is a dict-like node with attribute-access.
+
+        :param filename: The filename of the file.
+        :param file:     The file contents as string array.
+        """
+        self.filename = filename
+        self.file = file
+        self._file_nb = self.read_from_list(file)
+
+    @staticmethod
+    def read_from_list(file):
+        """
+        Reads a notebook from a string array and returns the NotebookNode
+        object.
+
+        :param file: The notebook file contents as JSON string array.
+        :return:     The notebook as NotebookNode.
+        """
+        return nbformat.reads(''.join(file), nbformat.NO_CONVERT)
+
+    def write_to_list(self, file_nb):
```

Comment:
I am not entirely sure what you mean by *a parameter that is retrieved from the class* but I had two things in mind:

1. the Bear works with a copy of `_file_nb` and when he has done his job, `file_nb` has changed, i.e. 
`file_nb != NotebookHandler(filename, file).file_nb`.
1. due to the issue with the final newline charater '\n' (cf. line 50) the `write_to_list` method needs to know how the original `file` looked like.

since I need both the new `file_nb` as well as the original `file` to properly construct the needed list (string array) I thought this is simplest solution. does this answer your question?

in any case, I have no objections against changing the approach :)
maybe rename the parameter to something like `file_nb_changed` to avoid confusion?